src/EventStore.Core/Services/Archiver/Storage/IArchiveStorage.cs

```diff
@@ -2,17 +2,39 @@
 // Event Store Ltd licenses this file to you under the Event Store License v2 (see LICENSE.md).
 
 using System;
+using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
+using EventStore.Core.Services.Archiver.Storage.Exceptions;
 
 namespace EventStore.Core.Services.Archiver.Storage;
 
 public interface IArchiveStorage {
-	public static IArchiveStorage None = new NoArchiveStorage();
-	public ValueTask StoreChunk(string path);
-}
+	/// <summary>Stores a chunk in the archive</summary>
+	/// <param name="chunkPath">The path of the chunk to archive</param>
+	/// <param name="ct">The token to monitor for cancellation requests</param>
+	/// <exception cref="ChunkDeletedException">Thrown if the chunk file is deleted while being archived</exception>
+	/// <exception cref="OperationCanceledException">Thrown if cancellation was requested</exception>
+	/// <returns>
+	/// <see langword="true"/> if the chunk was successfully archived<br/>
+	/// <see langword="false"/> if there was a recoverable error
+	/// </returns>
+	public ValueTask<bool> StoreChunk(string chunkPath, CancellationToken ct);
```

Comment:
It's not clear what the caller needs to do with **false** when returned? What is a recoverable error?

Reply:
the caller needs to retry after some time when `false` is returned. By recoverable error, I mean that there was a temporary issue storing the chunk (e.g if the storage server is temporarily down or if there is an issue with the network connection)

Reply:
i'll amend the code docs to clarify that, thanks

Reply:
i could be persuaded but im not sure about the return value to indicate retryable errors vs exceptions to indicate nonretryable, unless it's a common pattern i'd probably just handle it all through returnvalue or exceptions (probably exceptions in this case)